gateway.php

```diff
@@ -480,7 +480,7 @@ function create_url($order_id){
          **/
         function pesapal_xml($order_id) {
           
-          $order                      = new WC_Order( $order_id );
+          $order                      = new wc_get_order( $order_id );
           $pesapal_args['total']      = $order->get_total();
           $pesapal_args['reference']  = $order_id;
           $pesapal_args['first_name'] = $order->billing_first_name;
```

Comment:
We should probably also change this to `$order->get_billing_first_name()`. In WooCommerce 3.0+, we should not access these properties directly.